examples/cluster/main.tf

```diff
@@ -0,0 +1,33 @@
+terraform {
+  required_providers {
+    dfcloud = {
+      source = "registry.terraform.io/dragonflydb/dfcloud"
+    }
+  }
+}
+
+provider "dfcloud" {
+}
+
+resource "dfcloud_datastore" "cache-cluster" {
+  name = "frontend-cache-cluster"
+
+  location = {
+    region   = "us-central1"
+    provider = "gcp"
+  }
+
+  tier = {
+    max_memory_bytes = 6000000000
+    performance_tier = "dev"
+    replicas         = 1
+  }
+
+  cluster = true
+
+  shard_memory = 3000000000
```

Comment:
shard memory should be optional and the size should be verified to match the predefined sizes, is it?

Is this validation already happens in the backend irrespective of the terraform provider ?